packages/spectacle/src/components/slide-layout.tsx

```diff
@@ -195,11 +196,117 @@ const Quote = ({
     </Box>
   </Slide>
 );
+
+/**
+ * Generic Codepane utility with optional Description text
+ */
+const CodeLayout = ({
+  children,
+  language,
+  text,
+  textProps,
+  ...props
+}: Omit<SlideProps, 'children'> & {
+  children: string;
+  language: string;
+  text?: string | ReactNode;
+  textProps?: ComponentProps<typeof Text>;
+}) => (
+  <Box data-testid="CodePane">
+    {text ? (
+      <Text margin={8} {...textProps}>
+        {text}
+      </Text>
+    ) : null}
+    <CodePane language={language} {...props}>
+      {children}
+    </CodePane>
+  </Box>
+);
+
+/**
+ * single Code Pane with optional Title layout
+ */
+const SingleCodeLayout = ({
+  children,
+  language,
+  title,
+  titleProps,
+  ...rest
+}: Omit<SlideProps, 'children'> & {
+  children: string;
+  language: string;
+  title?: string | ReactNode;
+  titleProps?: ComponentProps<typeof Text>;
+}) => {
+  return (
+    <Slide {...rest}>
+      <FlexBox
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          height: '100%'
+        }}
+      >
+        {title ? <Heading {...titleProps}>{title}</Heading> : null}
+        <CodeLayout language={language} {...rest}>
+          {children}
+        </CodeLayout>
+      </FlexBox>
+    </Slide>
+  );
+};
+
+/**
+ * multiple Code Panes with optional Description, with optional Title layout
+ */
+const MultiCodeLayout = ({
+  codePaneProps,
+  title,
+  titleProps,
+  ...rest
+}: Omit<SlideProps, 'children'> & {
```

Comment:
I like this, but have a couple thoughts. 

First, I think `codePanProps` might lead to confusion – since they aren't really props for `CodePane` items. Maybe we can shorten that to something like `codeBlocks` or something? (Idk, naming is hard.)

Next, what do you think about offering a grid layout so a user could choose between side-by-side and vertically stacked?

![image](https://user-images.githubusercontent.com/12721310/187772256-5f06f5f0-395b-4d7e-83a2-9df9fe446591.png)

If you think that makes sense, here's what I used for that:

```tsx
const MultiCodeLayout = ({
  codePaneProps,
  title,
  titleProps,
  numColumns = 1,
  ...rest
}: Omit<SlideProps, 'children'> & {
  codePaneProps: {
    code: string;
    language: string;
    description?: string | ReactNode;
    descriptionProps?: ComponentProps<typeof Text>;
  }[];
  title?: string | ReactNode;
  titleProps?: ComponentProps<typeof Text>;
  numColumns?: number;
}) => {
  return (
    <Slide {...rest}>
      {title ? <Heading {...titleProps}>{title}</Heading> : null}
      <Grid
        gridRowGap={1}
        gridColumnGap={1}
        gridTemplateColumns={`repeat(${numColumns}, 1fr)`}
        maxWidth="100%"
      >
        {codePaneProps.map(
          ({ code, language, description, descriptionProps, ...props }, i) => (
            <CodeLayout
              key={i}
              text={description}
              textProps={descriptionProps}
              language={language}
              {...props}
            >
              {code}
            </CodeLayout>
          )
        )}
      </Grid>
    </Slide>
  );
};
```

I had to also delete [this line](https://github.com/FormidableLabs/spectacle/blob/605436120df0d5694d9a1210b9ea0d0675d5bc33/packages/spectacle/src/components/code-pane.tsx#L160) to get that working, but I think we should maybe have a fluid width for code panes anyways 🤔 

Happy to chat more about this! Let me know what you think!

Reply:
Glad you like it! I think option to use a grid layout is a good idea. I'll make the changes.

Reply:
> First, I think codePanProps might lead to confusion – since they aren't really props for CodePane items. 

These properties do get passed to the `CodePane` component which is why I thought naming them as such would reduce confusion. Leaving the name the same for now, but happy to continue the discussion.

Reply:
It looks like some of these props get passed through `CodeLayout` into `CodePane` – but not _all_ of these fields get passed all the way through to `CodePane`, and there are some `CodePane` props that don't exist here.

For example, having `descriptionProps` inside of `codePaneProps` feels a bit confusing to me, since we have an explicit prop type:

```ts
export type CodePaneProps = {
  children: string;
  language: string | undefined;
  theme?: Record<string, CSSProperties>;
  stepIndex?: number;
  highlightRanges?: Ranges;
  showLineNumbers?: boolean;
};
```

Reply:
Yes! You're exactly right and I agree it was confusing. I fixed this to use the exported Code Pane Props type and extended it for `CodeBlocks`